### PR TITLE
fix(autoware_shape_estimation): fix unusedFunction

### PR DESCRIPTION
**Pull request by kobayu858**
## Description
This is a fix based on cppcheck unusedFunction warnings.
I believe this is a false positive caused by the fact that it is defined in cpp but only used in hpp.

```
perception/autoware_shape_estimation/lib/corrector/utils.cpp:44:0: style: The function 'correctWithDefaultValue' is never used. [unusedFunction]
bool correctWithDefaultValue(
^

perception/autoware_shape_estimation/lib/corrector/utils.cpp:264:0: style: The function 'correctWithReferenceYaw' is never used. [unusedFunction]
bool correctWithReferenceYaw(
^

perception/autoware_shape_estimation/lib/corrector/utils.cpp:335:0: style: The function 'correctWithReferenceYawAndShapeSize' is never used. [unusedFunction]
bool correctWithReferenceYawAndShapeSize(
^
```
## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by kobayu858** [comment]
Perhaps if the entire definition were moved to hpp, this warning would go away.
However, since it was mentioned in the comments that refactoring is planned, I decided to suppress it this time.

**Comment by Shin-kyoto** [comment]
@yukkysaito 

Can you review this PR?
The TODO comment mentioned [here](https://github.com/autowarefoundation/autoware.universe/pull/8575#issuecomment-2304019397) is your comment, but we cannot understand the background of it.

```cpp
bool correctWithDefaultValue(
  const CorrectionBBParameters & param, autoware_perception_msgs::msg::Shape & shape,
  geometry_msgs::msg::Pose & pose)
{
  // TODO(Yukihiro Saito): refactor following code
```

**Comment by kobayu858** [comment]
@yukkysaito 

Please confirm the PR. Thank you very much.

**Comment by yukkysaito** [comment]
> The TODO comment mentioned https://github.com/autowarefoundation/autoware.universe/pull/8575#issuecomment-2304019397 is your comment, but we cannot understand the background of this.

What is the relationship between this question and the revisions in this PR?

**Review by yukkysaito — APPROVED** [review]

**Comment by kobayu858** [comment]
@yukkysaito 

As for the above comments, they can be ignored as the modification policy was changed yesterday.

**Comment by kobayu858** [comment]
I am checking to see if the warning can be resolved by moving the function definitions from the hpp file where these functions are used to cpp.

**Comment by kobayu858** [comment]
@yukkysaito @kaancolak @YoshiRi cc: @veqcc 

I suppressed the warning by moving the side that uses these functions from .hpp to .cpp.

Thank you in advance for your help and please check the PR again.

File: perception/autoware_shape_estimation/lib/corrector/reference_shape_size_corrector.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 Autoware Foundation. All rights reserved.
```
Comment (YoshiRi) [inline perception/autoware_shape_estimation/lib/corrector/reference_shape_size_corrector.cpp:1]:
```suggestion
// Copyright 2024 Tier IV, Inc.. All rights reserved.
```

File: perception/autoware_shape_estimation/lib/corrector/vehicle_corrector.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Autoware Foundation. All rights reserved.
```
Comment (YoshiRi) [inline perception/autoware_shape_estimation/lib/corrector/vehicle_corrector.cpp:1]:
```suggestion
// Copyright 2024 Tier IV, Inc.. All rights reserved.
```

**Review by YoshiRi — APPROVED** [review]
LGTM with trivial comments.

File: perception/autoware_shape_estimation/lib/corrector/reference_shape_size_corrector.cpp (hunk 1 above)
Comment (kobayu858) [reply perception/autoware_shape_estimation/lib/corrector/reference_shape_size_corrector.cpp]: Thank you very much. Comment has been fixed.

File: perception/autoware_shape_estimation/lib/corrector/vehicle_corrector.cpp (hunk 2 above)
Comment (kobayu858) [reply perception/autoware_shape_estimation/lib/corrector/vehicle_corrector.cpp]: Thank you very much. Comment has been fixed.

**Pull request merged**